In [1]:
import pandas as pd

In [2]:
# Load and label ChatGPT data
reddit_ChatGPT = pd.read_csv("Reddit_ChatGPT.csv")
reddit_ChatGPT["target"] = 1
reddit_ChatGPT["source_model"] = "chatgpt"

# Load and label Claude data
reddit_Claude = pd.read_csv("Reddit_Claude.csv")
reddit_Claude["target"] = 1
reddit_Claude["source_model"] = "claude"

# Load and label Gemini data
reddit_Gemini = pd.read_csv("Reddit_Gemini.csv")
reddit_Gemini["target"] = 1
reddit_Gemini["source_model"] = "gemini"

# Load and label Real/Human data
reddit_Real = pd.read_csv("Reddit_Real.csv")
reddit_Real["target"] = 0
reddit_Real["source_model"] = "human"

In [3]:
# Combine all four datasets
combined_df = pd.concat(
    [reddit_ChatGPT, reddit_Claude, reddit_Gemini, reddit_Real],
    ignore_index=True
)

# Shuffle rows to remove ordering bias
combined_df = combined_df.sample(frac=1, random_state=881).reset_index(drop=True)

In [4]:
combined_df.to_csv('Combined_Dataset.csv', index=False)
combined_df

,Title,URL,Score,Upvote_Ratio,Num_Comments,Post_Date,target,source_model
0,Protests break out in major cities over new ta...,https://www.reddit.com/r/politics/comments/pos...,666,0.90,4646,03-27-2026,1,gemini
1,ABC pulls Taylor Frankie Paul's season of 'Bac...,https://www.nbcnews.com/pop-culture/pop-cultur...,6345,0.96,666,2026-03-19,0,human
2,Researchers uncover unusual pattern in Arctic ...,https://www.apnews.com/article/arctic-sea-ice-...,3960,0.73,1497,2026-04-02,1,chatgpt
3,Heavy metal memorabilia on offer at Julien's '...,https://www.reuters.com/business/media-telecom...,2475,0.82,238,2026-03-19,1,chatgpt
4,Rays RHP Ryan Pepiot (hip) to begin season on ...,https://www.reuters.com/sports/rays-rhp-ryan-p...,2457,0.72,148,2026-03-15,1,chatgpt
...,...,...,...,...,...,...,...,...
1201,US federal jury convicts Assad-era Syrian offi...,https://www.reuters.com/world/us-federal-jury-...,148,0.91,20,2026-03-17,0,human
1202,Air Canada flight hits vehicle on runway at La...,https://edition.cnn.com/2026/03/23/us/laguardi...,1433,0.96,166,2026-03-23,0,human
1203,Florida bill against marrying your first cousi...,https://www.ctvnews.ca/world/article/florida-b...,3923,0.94,419,2026-03-17,0,human
1204,Exclusive: Iran wants 'serious review' of Gulf...,https://www.reuters.com/world/middle-east/iran...,601,0.93,128,2026-03-16,0,human


**Dataset Target Variable (`AI_Post`) Distribution**

In [5]:
combined_df['target'].value_counts()

target
0    606
1    600
Name: count, dtype: int64

Classes are nearly perfectly balanced:
  - 606 Real (`AI_Post` = 0) Reddit Posts
  - 600 Fake (`AI_Post` = 1) Reddit Posts (200 - ChatGPT, 200 - Claude, 200 - Gemini)